### ISSUE TITLE
[BUG] [module:classification]

### ISSUE THREAD

**Issue by jmwhyte** [sktime/sktime#5702]
**Describe the bug**

**To Reproduce**

```python
<Paste your code here>
```

**Expected behavior**

**Additional context**

**Versions**
<details>

</details>

**Comment by fkiraly** [comment]
@jmwhyte, here's an update on this (which has triggered some `random_state` handling thoughts), and questions on the design:

* `sklearn`, natively, does the same - but with the same problems as you highlight.
* to handle dependent random states, we've introduced some functionality in `scikit-base` now: https://github.com/sktime/skbase/issues/279 (open discussion), https://github.com/sktime/skbase/pull/268 (PR)

The latter would allow to add dependent random state handling in tuners. My current thinking is as follows:

* new parameter `random_state_handling`, which defaults to current, but has a setting where internal tuned clones are sampled dependent `random_state` parameters
* the classification grid search cannot be directly changed due to its direct dependency on `sklearn` which does not have this random state handling - so it ought to be rebuilt based on the forecasting tuners.

**Comment by fkiraly** [comment]
FYI @ericjb - I am not sure whether to handle random state setting in `clone`, or externally.

option 1: 

`tuned_est_i = self.clone(random_state="new").set_params(**grid_params)`

Here, `clone` would have to cause side effects on self, to set a counter how often the estimator already had been cloned, or similar.

option 2:

Clones are generated as current, but `set_random_seed` is called after `sample_dependent_seed`.


One problem in both, `sample_dependent_seed` and `set_random_seed` does things in a chain. So, the second clone within the tuner gets the same random seed as the first component in the first clone.

A more general version of this problem is the tree-like structure of the dependencies, and the strictly sequential strategy in generating dependent random seeds.

**Comment by jmwhyte** [comment]
Thanks for the updates, I'm about to go on leave, but look forward to thinking through the notes when I return. I have a little more project time for grid searching, so might get to try out some ideas.

**Comment by fkiraly** [comment]
Yes, I've been getting deep into the woods in two directions - tuning backends and random seed management. Let me know when you are back to `sktime`, and we can draw up a plan.
